pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -241,7 +241,9 @@ func (s *Scheduler) calculateExistingMachines(stateNodes []*state.Node, daemonSe
 		// We don't use the status field and instead recompute the remaining resources to ensure we have a consistent view
 		// of the cluster during scheduling.  Depending on how node creation falls out, this will also work for cases where
 		// we don't create Node resources.
-		s.remainingResources[node.Node.Name] = resources.Subtract(s.remainingResources[node.Node.Name], node.Capacity())
+		if _, ok := s.remainingResources[node.Node.Labels[v1alpha5.ProvisionerNameLabelKey]]; ok {
```

Comment:
What situation does the new check handle?

Reply:
This isn't a new check. The previous changes broke this path (this was originally the Provisioner label for existing machines/nodes). Since we will now build existing machines for all nodes we own (including ones that might not have a provisioner label, thinking about the future for manually launched machines, we need to check that the provisioner label:
1. Exists
2. Has a label mapping inside or remaining resources so that we only subtract when it does
3.